toolbox/api/datagalaxy_api_dictionary.py

```diff
@@ -0,0 +1,162 @@
+import logging
+
+import requests as requests
+from toolbox.api.datagalaxy_api_workspaces import DataGalaxyBulkResult, to_bulk_tree
+
+
+class DataGalaxyApiDictionary:
+    def __init__(self, url: str, access_token: str, workspace: dict):
+        self.url = url
+        self.access_token = access_token
+        self.workspace = workspace
+
+    def list_sources(self, workspace_name: str) -> list:
+        if self.workspace["isVersioningEnabled"]:
```

Comment:
It might be better to move this logic in the constructor. We don't want to build an "invalid" `DataGalaxyApiDictionary`. If the workspace is versioned then we should fail fast.